src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,110 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { Loader } from '../Loader';
+import { ErrorsMessages } from '../../types/ErrorsMessages';
+
+type Props = {
+  todo: Todo,
+  handleChecker: (id: number, data: unknown) => void,
+  removeTodo: (id: number) => void,
+  processings: number[],
+  errorMessage: (message: ErrorsMessages) => void
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  handleChecker,
+  removeTodo,
+  processings,
+  errorMessage,
+}) => {
+  const { id, completed, title } = todo;
+  const [isFormActive, setIsFormActive] = useState(false);
+  const [inputValue, setInputValue] = useState(title);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    setInputValue(event.target.value);
+  };
+
+  const validateTitle = () => {
+    if (!inputValue.trim().length) {
+      removeTodo(id);
+      errorMessage(ErrorsMessages.Title);
+
+      return null;
+    }
+
+    if (title === inputValue) {
+      setIsFormActive(false);
+
+      return null;
+    }
+
+    setIsFormActive(false);
+    handleChecker(id, { title: inputValue });
```

Comment:
better to move this `handleChecker` function outside `validateTitle`, `validateTitle` seems only to validate and `handleChecker` to update data, you can create one common function to call `validateTitle` and then `handleChecker`